elki/src/main/java/de/lmu/ifi/dbs/elki/math/statistics/dependence/McdeMwpDependenceMeasure.java

```diff
@@ -0,0 +1,227 @@
+
+        final double U = R - ((double)(n1 * (n1 - 1))) / 2;
+        final long n2 = cutLength - n1;
+
+        final long two_times_sqrt_max_long = 6074000999L;
+        if(n1 + n2 > two_times_sqrt_max_long)
+            throw new AbortException("Long type overflowed. Too many objects: Please subsample and try again with smaller data set.");
+
+        final double b_end = corrected_ranks[(end-1)].correction;
+        final double b_start = start == 0 ? 0 : corrected_ranks[(start-1)].correction;
+        final double correction = (b_end - b_start) / (cutLength * (cutLength -1));
+        final double std = Math.sqrt(( ((double) (n1 * n2)) / 12) * (cutLength + 1 - correction));
+
+        if(std == 0) return 0;
+        else{
+            final double mean = ((double) (n1 * n2)) / 2;
+            final double Z = Math.abs((U - mean) / std);
+            return erf(Z / Math.sqrt(2)); // Note that this is equivalent to do 1-2*(1-cdf(Z,0,1));
+            // erf(Z / Math.sqrt(2)) is the cdf of the half-normal distribution
+        }
+    }
+
+    /**
+     * Parameterization class.
+     *
+     * @author Alan Mazankiewicz
+     * @author Edouard Fouché
+     */
+    public static class Parameterizer extends AbstractParameterizer {
+
+        /**
+         * Parameter that specifies the number of iterations in the Monte-Carlo
+         * process of identifying high contrast subspaces.
+         */
+        public static final OptionID M_ID = new OptionID("McdeMwp.m", "No. of Monte-Carlo iterations.");
+
+        /**
+         * Parameter that specifies the size of the slice
+         */
+        public static final OptionID ALPHA_ID = new OptionID("McdeMwp.alpha", "Expected share of instances in slice (independent dimensions).");
+
+        /**
+         * Parameter that specifies the size of the marginal restriction. Note that in the original paper
+         * alpha = beta and as such there is no explicit distinction between the parameters.
+         */
+        public static final OptionID BETA_ID = new OptionID("McdeMwp.beta", "Expected share of instances in marginal restriction (dependent dimensions).");
+
+        /**
+         * Parameter that specifies the random seed.
+         */
+        public static final OptionID SEED_ID = new OptionID("McdeMwp.seed", "The random seed.");
+
+        /**
+         * Holds the value of {@link #M_ID}.
+         */
+        protected int m = 50;
+
+        /**
+         * Holds the value of {@link #ALPHA_ID}.
+         */
+        protected double alpha = 0.5;
+
+        /**
+         * Holds the value of {@link #BETA_ID}.
+         */
+        protected double beta = 0.5;
+
+        /**
+         * Random generator.
+         */
+        protected RandomFactory rnd;
+
+        @Override
+        protected void makeOptions(Parameterization config) {
+            super.makeOptions(config);
+
+            final IntParameter mP = new IntParameter(M_ID, 50) //
```

Comment:
Parameters such as M_ID should usually go to the class where they are used, which here is the abstract parent class; and then should be inherited via the `super.makeOptions();` call. This way they won't be duplicated in multiple subclasses.

Reply:
Not relevant anymore